### PR TITLE
merge upstream changes from `v0.13.0`

### PR DESCRIPTION
**Pull request by antoinerg**
Which brings along a few security patches such as a fix for CVE-2023-49569

I had to resolve a few conflicts in `buildpack.toml`. I kept all the current changes from `5.4-release` although upstream brings in new versions of Python which we will miss out on.

See https://github.com/plotly/paketo-buildpacks_cpython/compare/upstream-1.13.0..paketo-buildpacks:cpython:v1.13.0 for a diff between upstream's `v1.13.0` and the content of the current branch.

I published the new image with:
```
./scripts/package.sh --version 5.4.0b2 --output quay.io/plotly/paketo-buildpacks-cpython:5.4.0b2
```

### CODE REVIEW

**Comment by BRONSOLO** [comment]
same here re: the release branch -- should this be landing against main with the idea being we'd cut the 5.4 release branch when we freeze off of main?

File: build.go [hunk 1]
```diff
@@ -38,6 +39,11 @@ type PythonInstaller interface {
 	) error
 }
 
+// PythonPipCleanup defines the interface for cleaning up pip after a python installation
+type PythonPipCleanup interface {
+	Cleanup(packages []string, targetLayer string) error
+}
+
```
Comment (tarzzz) [inline build.go:46]: Does "cleaning up" pip means deleting `pip`? If so it wouldn't be a good idea for workspaces as we need `pip`

**Comment by antoinerg** [comment]
> @antoinerg Left a comment around `pipCleanup` -- Are you able to test this properly on both workspaces and dash apps?
> 
> Making sure workspace have all the dependencies installed (pip, apt deps etc.) and predeploy scripts continue to work.

I added this as a TODO in https://github.com/plotly/dek-deployment/pull/3963. Hopefully, I can test this out this afternoon.

**Comment by antoinerg** [comment]
> I added this as a TODO in [plotly/dek-deployment#3963](https://github.com/plotly/dek-deployment/pull/3963). Hopefully, I can test this out this afternoon.

@tarzzz please see https://github.com/plotly/dek-deployment/pull/3963 for test updates. I provided a cluster where you can login and try to break things!

**Comment by antoinerg** [comment]
I think this one is ready for another round of review!

**Comment by antoinerg** [comment]
I got an approval over Slack!

**Pull request merged**